src/index.ts

```diff
@@ -86,6 +113,9 @@ export interface Event<D extends EventData, C extends EventContext<any>> {
 }
 
 export type Aggregate<A extends any[], T> = (...args: A) => Promise<Option<T>>;
+export type AggregateWhile<A extends any[], T> = (...args: A) =>
+  Promise<(whileTrue: (event: any) => boolean, guid: string) => Promise<Option<T>>>;
```

Comment:
Should this be `guid`? Why not make `whileTrue` a more generic function?

Reply:
Yeah, I think I should definitely make `guid` a generic optional argument like `args`. 
`WhileTrue` is probably as generic as it can be: while the function returns true, the aggregator continues..

Reply:
You could make it a function with signature:
```Typescript
export type whileTrue<K>(args: { event: Event<any, any, any>, condition: K}) => Promise<Option<T>>
``` ?

Reply:
In the guid example `K = string`

Reply:
OK yep I see what you mean. Thanks @DennisSchwartz !